embedding-calculator/src/services/facescan/plugins/base.py

```diff
@@ -71,11 +71,19 @@ def _download(cls, url: str, output):
     def _extract(self, filename: str):
         os.makedirs(self.path, exist_ok=True)
         with ZipFile(filename, 'r') as zf:
-            for info in zf.infolist():
-                if info.is_dir():
-                    continue
-                file_path = Path(self.path) / Path(info.filename).name
-                file_path.write_bytes(zf.read(info))
+            if str(self.path)[-18:] == 'face_mask_detector':
```

Comment:
Add a parameter "retain_folder_structure" to BasePlugin, and add it to every model initialization instead of this.
And change the behavior using this variable